src/main/java/io/github/oliviercailloux/j_voting/preferences/interfaces/MutablePreference.java

```diff
@@ -1,27 +1,34 @@
 package io.github.oliviercailloux.j_voting.preferences.interfaces;
 
+import com.google.common.graph.ImmutableGraph;
 import com.google.common.graph.MutableGraph;
 
 import io.github.oliviercailloux.j_voting.Alternative;
 
 /**
  * A mutable preference is a preference in which some alternatives can be added,
  * which means some edges (pair of alternatives) can be added.
- *
- * If an alternative a is strictly better than an alternative b,
- * it is impossible to change this afterwards.
+ * 
+ * If an alternative a is strictly better than an alternative b, it is
+ * impossible to change this afterwards.
  */
 public interface MutablePreference extends Preference {
-    
-    /**
-     * The returned graph reads through this object: if this object is mutable,
-     * any modification to this object modifies the returned graph,
-     * and conversely.
-     *
-     * {@inheritDoc}
-     *
-     * This graph is mutable.
-     */
-    @Override
-    public MutableGraph<Alternative> asGraph();
+
+	/**
+	 * The returned graph can't be modified, though is not necessarily immutable.
+	 *
+	 * @see MutablePreference#asMutableGraph()
+	 */
+	@Override
+	public ImmutableGraph<Alternative> asGraph();
```

Comment:
“is not necessarily immutable”, dit la doc, donc renvoyer `Graph` et pas `ImmutableGraph` !

Il faut supprimer cette méthode de cette interface et déplacer cette documentation sur `Preference#asGraph()`.